core/src/init.rs

```diff
@@ -0,0 +1,62 @@
+use crate::config::read_config;
+use crate::config::yaml::TaskConfigYaml;
+
+use anyhow::Result;
+use futures::StreamExt;
+use nix::{
+    libc::{SIGABRT, SIGCHLD, SIGHUP, SIGPIPE, SIGTERM, SIGTSTP},
+    sys::wait::waitpid,
+    unistd::Pid,
+};
+use signal_hook::{iterator::exfiltrator::WithOrigin, low_level::siginfo::Origin};
+use signal_hook_async_std::SignalsInfo;
+use std::env;
+use tracing::info;
+
+use crate::task::{ContextMap, Task};
+
+const SIGS: &[i32] = &[SIGABRT, SIGTERM, SIGCHLD, SIGHUP, SIGPIPE, SIGTSTP];
+
+pub struct Alfad {
+    pub builtin: Vec<TaskConfigYaml>,
+}
+
+impl Alfad {
+    pub fn run(self) -> Result<()>{
+        let mut signals = SignalsInfo::<WithOrigin>::new(SIGS).unwrap();
+
+        smol::spawn(async move {
+            while let Some(sig) = signals.next().await {
+                match sig {
+                    Origin {
+                        signal: SIGCHLD,
+                        process: Some(proc),
+                        ..
+                    } => {
+                        // Ignore Err(_) since ECHILD is expected
+                        waitpid(Some(Pid::from_raw(proc.pid)), None).ok();
+                    }
+                    _ => {}
+                }
+            }
+        })
+        .detach();
+
+        env::set_var("SMOL_THREADS", "8");
```

Comment:
I would do that _only_ if `SMOL_THREADS` was not defined yet from the outside.

Reply:
I'll probably switch to a custom Executor anyway so this is just for testing purposes